### ISSUE TITLE
Last User should not be root

### ISSUE THREAD

**Issue by gcallaghan** [hadolint/hadolint#184]
- [] This is a bug report
- [X ] This is a feature request
- [ x] I searched existing issues before opening this one

### Expected behavior
Last `USER` should not be root.

It's common to need to switch to root to install additional packages, however, there should be a layer that sets the user to non-root afterward.

**Comment by grayside** [comment]
I've had to suppress DL3002 in my execution because it's naive about any switch to root being bad.

**Comment by grayside** [comment]
My interpretation of the report is to soften an existing rule without losing the main security concern. Have I misunderstood?

**Comment by lorenzo** [comment]
@grayside I ws more interested in the particular reasons you had to disable the rule

**Comment by gcallaghan** [comment]
Apologies. Re-reading, I see that my feature request is unclear.

I would like to have the ability to switch to root and back again without a lint error.
If the last user is root, I'd like a lint error to be thrown.

Currently, my only option is to disable DL3002 as @grayside mentioned. Ignoring the error as a warning has the same effect as developers in my org would then get trained to ignore warnings. 😢

**Comment by zymergen-mbarton** [comment]
I agree. In combination with not using `sudo` in DL3004, there's no way to do root operations such as installing additional packages without getting a lint error.

**Comment by gurvindersingh** [comment]
I am having the same issues and need to ignore the rule to install software as root. It would be good to have a check which raises the issue if last user is root otherwise not.

**Comment by vquemener** [comment]
I am still experiencing this issue :
```bash
$ cat Dockerfile 
FROM busybox:1
USER root
RUN echo
USER guest
$ docker run --rm -i hadolint/hadolint < Dockerfile
/dev/stdin:2 DL3002 Last USER should not be root
$ docker run --rm -i hadolint/hadolint /bin/hadolint --version
Haskell Dockerfile Linter v1.10.3-1-g69be604
```
I feel like something is wrong with this [unit test](https://github.com/hadolint/hadolint/blob/master/test/Spec.hs#L94-L101) : should it not be `ruleCatchesNot` instead of `ruleCatches`?

**Comment by lorenzo** [comment]
heh, that's funny... thanks for reporting.

**Comment by vquemener** [comment]
Thanks for fixing this!
